### PR TITLE
KAS-2742: Updated newsletter model

### PR DESCRIPTION
**Pull request by sergiofenoll**
Updates the news item model and moves it from a lisp file to a JSON file, also updates all the necessary links & configs.

### CODE REVIEW

File: config/resources/newsletter-domain.json [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+        },
+        "themes": {
+          "target": "theme",
+          "predicate": "dct:subject",
+          "cardinality": "many"
+        },
+        "agenda-item-treatments": {
+          "target": "agenda-item-treatment",
+          "predicate": "prov:wasDerivedFrom",
+          "cardinality": "many"
```
Comment (erikap) [inline config/resources/newsletter-domain.json]: See discussion in https://github.com/kanselarij-vlaanderen/frontend-kaleidos/pull/1561#discussion_r1046195169. Must be a `"one"` relation.

File: config/resources/newsletter-domain.json [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+      "relationships": {
+        "newsletters": {
+          "target": "news-item",
+          "predicate": "dct:subject",
+          "cardinality": "many",
+          "inverse": true
+        }
+      },
+      "features": [
+        "no-pagination-defaults",
```
Comment (erikap) [inline config/resources/newsletter-domain.json]: Can we get rid of this one by loading the themes in frontend like we load other codelists?

File: config/resources/newsletter-domain.json [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+          "target": "meeting",
+          "predicate": "ext:heeftMailCampagnes",
+          "cardinality": "one",
+          "inverse": true
+        }
+      },
+      "features": [
+        "include-uri"
+      ],
+      "new-resource-base": "http://themis.vlaanderen.be/id/mailcampagne"
```
Comment (erikap) [inline config/resources/newsletter-domain.json]: Trailing `/` is missing

**Review by erikap — CHANGES_REQUESTED** [review]

File: config/resources/newsletter-domain.json [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+          "predicate": "nie:htmlContent"
+        },
+        "plain-text": {
+          "type": "string",
+          "predicate": "prov:value"
+        },
+        "publication-date": {
+          "type": "datetime",
+          "predicate": "dct:issued"
+        },
```
Comment (ValenberghsSven) [inline config/resources/newsletter-domain.json]: We don't use this relation in our app.
`Themis-export-service` inserts it.
better to remove?

File: config/resources/newsletter-domain.lisp [hunk 5]
```diff
@@ -1,50 +0,0 @@
-(define-resource newsletter-info ()
-  :class (s-prefix "besluitvorming:NieuwsbriefInfo")
-  :properties `((:text                  :string   ,(s-prefix "besluitvorming:inhoud"))
-                (:richtext              :string   ,(s-prefix "ext:htmlInhoud"))
-                (:subtitle              :string   ,(s-prefix "dbpedia:subtitle"))
-                (:mandatee-proposal     :string   ,(s-prefix "ext:voorstelVan"))
```
Comment (ValenberghsSven) [inline config/resources/newsletter-domain.lisp:6]: 5000 hits in data for `ext:voorstelVan`. predicate not cleaned up (that I could find).
If we keep the data, we need to mention it somewhere?

**Review by ValenberghsSven — COMMENTED** [review]
There are 66140 newsitems currently
65522 of those records have `prov:generated`

64965 records are propagated (same for all graphs)

Do we have to figure out what those 600 records are that are not connected to an `agendaItemTreatment` before migrating?

The other 600 that have a treatment but are not propagated can't all be from active agendas but they should get migrated just the same (I think)

**Comment by sergiofenoll** [comment]
Removing the migration service and rewriting the migration as separate migrations per predicate. That doesn't time out Virtuoso and removes a cumbersome extra service. These are the queries I used to verify that the number of news items match after running the migration on a recent-ish ACC database.

Before migration:
```sparql
PREFIX besluitvorming: <http://data.vlaanderen.be/ns/besluitvorming#>
PREFIX dbpedia: <http://dbpedia.org/ontology/>
PREFIX dct: <http://purl.org/dc/terms/>
PREFIX ext: <http://mu.semte.ch/vocabularies/ext/>
PREFIX nie: <http://www.semanticdesktop.org/ontologies/2007/01/19/nie#>
PREFIX prov: <http://www.w3.org/ns/prov#>

SELECT
(COUNT(DISTINCT ?nieuwsbericht) AS ?countNieuwsberichten)
(COUNT(DISTINCT ?subtitle) AS ?countSubtitle)
(COUNT(DISTINCT ?htmlContent) AS ?countHtmlContent)
(COUNT(DISTINCT ?plainText) AS ?countPlainText)
(COUNT(DISTINCT ?remark) AS ?countRemark)
(COUNT(DISTINCT ?modified) AS ?countModified)
(COUNT(DISTINCT ?agendaItemTreatment) AS ?countAgendaItemTreatment)
(COUNT(DISTINCT ?attachments) AS ?countAttachments)
WHERE {
  ?nieuwsbericht a besluitvorming:NieuwsbriefInfo .
  OPTIONAL { ?nieuwsbericht dbpedia:subtitle ?subtitle }
  OPTIONAL { ?nieuwsbericht ext:htmlInhoud ?htmlContent }
  OPTIONAL { ?nieuwsbericht besluitvorming:inhoud ?plainText }
  OPTIONAL { ?nieuwsbericht ext:opmerking ?remark }
  OPTIONAL { ?nieuwsbericht ext:aangepastOp ?modified }
  OPTIONAL { ?nieuwsbericht ^prov:generated ?agendaItemTreatment }
  OPTIONAL { ?nieuwsbericht ext:documentenVoorPublicatie ?attachments }
}
```
Results:
image


After migration: 
```sparql
PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX ex: <http://example.org/>
PREFIX besluitvorming: <http://data.vlaanderen.be/ns/besluitvorming#>
PREFIX dbpedia: <http://dbpedia.org/ontology/>
PREFIX dct: <http://purl.org/dc/terms/>
PREFIX ext: <http://mu.semte.ch/vocabularies/ext/>
PREFIX nie: <http://www.semanticdesktop.org/ontologies/2007/01/19/nie#>
PREFIX prov: <http://www.w3.org/ns/prov#>

SELECT
(COUNT(DISTINCT ?nieuwsbericht) AS ?countNieuwsberichten)
(COUNT(DISTINCT ?subtitle) AS ?countSubtitle)
(COUNT(DISTINCT ?htmlContent) AS ?countHtmlContent)
(COUNT(DISTINCT ?plainText) AS ?countPlainText)
(COUNT(DISTINCT ?remark) AS ?countRemark)
(COUNT(DISTINCT ?modified) AS ?countModified)
(COUNT(DISTINCT ?agendaItemTreatment) AS ?countAgendaItemTreatment)
(COUNT(DISTINCT ?attachments) AS ?countAttachments)
WHERE {
  ?nieuwsbericht a ext:Nieuwsbericht .
  OPTIONAL { ?nieuwsbericht dct:alternative ?subtitle }
  OPTIONAL { ?nieuwsbericht nie:htmlContent ?htmlContent }
  OPTIONAL { ?nieuwsbericht prov:value ?plainText }
  OPTIONAL { ?nieuwsbericht rdfs:comment ?remark }
  OPTIONAL { ?nieuwsbericht dct:modified ?modified }
  OPTIONAL { ?nieuwsbericht prov:wasDerivedFrom ?agendaItemTreatment }
  OPTIONAL { ?nieuwsbericht besluitvorming:heeftBijlage ?attachments }
}
```
Results:
image

Note that the difference in count of subtitles is due to a quirk in the data for older news items. I checked with Johan and the conclusion was that we could ignore that.

File: config/resources/newsletter-domain.lisp (hunk 5 above)
Comment (erikap) [reply config/resources/newsletter-domain.lisp:6]: Ik heb een story gemaakt om dit op te volgen: https://kanselarij.atlassian.net/browse/KAS-3802

**Review by erikap — APPROVED** [review]

**Pull request merged**